### ISSUE TITLE
Error when importing drafts

### ISSUE THREAD

**Issue by klamping** [lmammino/ghost2hexo#1]
I get an error when importing a JSON file that has drafts:

```
/Users/klamping/.nvm/versions/node/v8.9.0/lib/node_modules/ghost2hexo/src/index.js:78
    : path.join(dest, `${post.published_at.substr(0, 10)}_${post.slug}.md`)
                                           ^

TypeError: Cannot read property 'substr' of null
    at createPost (/Users/klamping/.nvm/versions/node/v8.9.0/lib/node_modules/ghost2hexo/src/index.js:78:44)
```

I'm pretty sure this is due to drafts having their "published_at" date set to null.

For reference this is the general structure of the draft JSON
```json
{
        "id": "someid",
        "uuid": "someid",
        "title": "Post title",
        "slug": "-slug",
        "feature_image": null,
        "featured": 0,
        "page": 0,
        "status": "draft",
        "locale": null,
        "visibility": "public",
        "meta_title": null,
        "meta_description": null,
        "author_id": "asdfasd",
        "created_at": "2014-11-21T23:46:49.000Z",
        "created_by": "59a1bf12b174ad0c69ae8874",
        "updated_at": "2014-11-21T23:46:49.000Z",
        "updated_by": "59a1bf12b174ad0c69ae8874",
        "published_at": null,
        "published_by": null,
        "custom_excerpt": null,
        "codeinjection_head": null,
        "codeinjection_foot": null,
        "og_image": null,
        "og_title": null,
        "og_description": null,
        "twitter_image": null,
        "twitter_title": null,
        "twitter_description": null
      }
```

My Ghost version is 1.7.1

**Comment by lmammino** [comment]
Judging from the error stack, It seems that this is happening because your `published_at` is null. That property is used to define files/folders structure. As you underlined, this is probably ghost's way to mark draft posts...

One way we could solve this is to change the way the filename are generated, (lines [73-76](https://github.com/lmammino/ghost2hexo/blob/master/src/index.js#L73-L76)) something like:

```javascript
const createPost = (post, dest, authorsMap) => {
  const destFile = post.page
    ? path.join(dest, '..', `${post.slug}.md`)
    : path.join(dest, `${post.published_at ? post.published_at.substr(0, 10) : '__draft__'}_${post.slug}.md`)
```

What do you think?

I am not using hexo anymore, so I am not sure this solution would make sense in this context.

**Comment by klamping** [comment]
Yeah, that's essentially what I did to my local copy. Didn't have enough time to really test it though. 

I'd say just leave this issue here and if anyone else runs across it and are interested enough they can work on a more polished solution.

**Comment by docrinehart** [comment]
Bumping into a similar issue with unpublished drafts. Might look at digging in and submitting a PR if I find the time, but alas, I have other priorities at the moment.

```
/Users/docrinehart/.nvm/versions/node/v11.4.0/lib/node_modules/ghost2hexo/src/index.js:79
      `${post.published_at.toString().substr(0, 10)}_${post.slug}.md`
                           ^

TypeError: Cannot read property 'toString' of null
    at createPost (<xxxxx>/node_modules/ghost2hexo/src/index.js:79:28)
    at posts.forEach.post (<xxxxx>/node_modules/ghost2hexo/src/index.js:110:23)
    at Array.forEach (<anonymous>)
    at Object.<anonymous> (<xxxxx>/node_modules/ghost2hexo/src/index.js:110:7)
    at Module._compile (internal/modules/cjs/loader.js:723:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:734:10)
    at Module.load (internal/modules/cjs/loader.js:620:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:560:12)
    at Function.Module._load (internal/modules/cjs/loader.js:552:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:776:12)
```

**Comment by docrinehart** [comment]
Following up: I tried applying the fix to my local copy to handle drafts, and the tool successfully processed the posts to markdown files, however, the content was missing in all the posts. There may be some schema differences in the most recent versions of Ghost and/or Hexo. 

Using Hexo 3.8.0, Ghost Export 2.11.1

**Comment by lmammino** [comment]
@docrinehart, do you have any other field that might contain the equivalent value (assuming Ghost schema changes)? Maybe we can get away with some ternary condition.

**Comment by docrinehart** [comment]
Yeah. I dug in a bit but haven't had a chance to reply. Ghost moved from
markdown to the mobiledoc format. I inspected the json, and there's
actually 3 available properties for the content in my case. Mobiledoc,
html, and plaintext. I replaced the content propery with html and at a
glance everything seemed to come over fine. I was going to take a closer
look this afternoon or Monday morning to inspect each post.

That said, html markup isn't ideal but is usable as-is. A better option
might be to look at mobiledoc to markdown processing, I think there's a
library that can do this already from my initial searches, but I haven't
dug further yet.

~Doc
